src/remap/function/parse_grok.rs

```diff
@@ -0,0 +1,162 @@
+use remap::prelude::*;
+use std::collections::BTreeMap;
+use std::sync::Arc;
+
+#[derive(Clone, Copy, Debug)]
+pub struct ParseGrok;
+
+impl Function for ParseGrok {
+    fn identifier(&self) -> &'static str {
+        "parse_grok"
+    }
+
+    fn parameters(&self) -> &'static [Parameter] {
+        &[
+            Parameter {
+                keyword: "value",
+                accepts: |v| matches!(v, Value::Bytes(_)),
+                required: true,
+            },
+            Parameter {
+                keyword: "pattern",
+                accepts: |v| matches!(v, Value::Bytes(_)),
+                required: true,
+            },
+        ]
+    }
+
+    fn compile(&self, mut arguments: ArgumentList) -> Result<Box<dyn Expression>> {
+        let value = arguments.required_expr("value")?;
+
+        let patternbytes = arguments
+            .required_literal("pattern")?
+            .as_value()
+            .clone()
+            .try_bytes()?;
+
+        let patternstr = String::from_utf8_lossy(&patternbytes).into_owned();
+
+        let mut grok = grok::Grok::with_patterns();
+        let pattern = Arc::new(grok.compile(&patternstr, true).map_err(|e| e.to_string())?);
+
+        Ok(Box::new(ParseGrokFn { value, pattern }))
+    }
+}
+
+#[derive(Debug, Clone)]
+struct ParseGrokFn {
+    value: Box<dyn Expression>,
+    // Wrapping pattern in an Arc, as cloning the pattern could otherwise be expensive.
```

Comment:
I'm actually not seeing us clone this value anywhere? Is this a left-over from a previous iteration? Can we remove the `Arc`?

Reply:
Ah wait, you're talking about cloning `ParseGrokFn` itself. We actually don't clone it at runtime, so there's not really a need for the `Arc`.

I'd personally vote to remove it, unless we actually have a benchmark that shows that we definitely need this.

Reply:
So `Expression` requires the struct to be `Clone`.  This is just to fullfill that requirement. I think it is used in various other components, not the `remap` transform, that need to clone?

Reply:
Ah yeah, some of the recent component changes required transforms to be `Clone`, I remember. 👍